src/bin/pgcopydb/copydb_schema.c

```diff
@@ -721,9 +710,13 @@ copydb_prepare_table_specs_hook(void *ctx, SourceTable *source)
 	 * When the Table Access Method used is not "heap" we don't know if the
 	 * CTID range scan is supported (see columnar storage extensions), so
 	 * we skip partitioning altogether in that case.
+	 *
+	 * Also, we cannot execute ANALYZE on a database that is in recovery mode,
+	 * so we skip partitioning in that case too.
```

Comment:
I think we should assume that the statistics are up-to-date when hitting a read-only source database, or at least have support for COPY partitioning in that case. An option to disable COPY partitioning would then be good to have for users who want to opt-out of single-table concurrency. This option would be useful in more cases.